src/postings/recorder.rs

```diff
@@ -304,6 +304,14 @@ impl Recorder for TfAndPositionRecorder {
     }
 }
 
+fn diff(val1: u32, val2: u32) -> u32 {
```

Comment:
This is not comparable to #1439 which just changed how the value is computed but produced the same value in any case (due to the `* -1.0` in the second branch). This function here would yield a different value for `delta_position` than intended.

I suspect the fix in deeper in the stack if `VInt32Reader` is expected to yield monotonically increasing positions but does not.